### PR TITLE
drop support of 0.10, 0.12 node (tests)

### PR DESCRIPTION
**Pull request by miherlosev**
@LavrovArtem 
Changes:
- [travis] move up `test-server`  task to the node 4.x
- rewrite server tests, playground and `gulpfile.js` using es6 syntax

### CODE REVIEW

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
FPR

File: src/.babelrc [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
    "compact": false,
-   "presets": [
-     "es2015-hammerhead",
-     "babel-preset-stage-2"
-   ],
+  "presets": [
+    "es2015-hammerhead",
+    "babel-preset-stage-2"
+  ],
```
Comment (LavrovArtem) [inline src/.babelrc]: Wrong indents

File: test/playground/server.js [hunk 2]
```diff
@@ -1,65 +1,58 @@
...
-    session._getPayloadScript = function () {
-        return '';
-    };
+function createSession () {
+    const session = new Session('test/playground/upload-storage');
 
-    session.getAuthCredentials = function () {
+    session._getIframePayloadScript = () => '';
+    session._getPayloadScript       = () => '';
+    session.getAuthCredentials      = () => {
```
Comment (LavrovArtem) [inline test/playground/server.js]: ```() => ({})```

File: test/playground/server.js [hunk 3]
```diff
@@ -1,65 +1,58 @@
...
         return {};
     };
-
-    session.handleFileDownload = function () {
+    session.handleFileDownload      = () => {
```
Comment (LavrovArtem) [inline test/playground/server.js]: ```() => void 0```

File: test/playground/server.js [hunk 4]
```diff
@@ -1,65 +1,58 @@
...
     };
-
-    session.handlePageError = function () {
+    session.handlePageError         = () => {
```
Comment (LavrovArtem) [inline test/playground/server.js]: ```() => void 0```

File: test/server/auth-test.js [hunk 5]
```diff
@@ -1,56 +1,49 @@
...
         session = new Session();
 
-        session.getAuthCredentials = function () {
-            return null;
-        };
-
-        session.handleFileDownload = function () {
-        };
+        session.getAuthCredentials = () => null;
+        session.handleFileDownload = () => {};
```
Comment (LavrovArtem) [inline test/server/auth-test.js]: ```() => void 0```

File: test/server/external-proxy-test.js [hunk 6]
```diff
@@ -1,66 +1,56 @@
...
+    before(() => {
         session = new Session();
 
-        session.getAuthCredentials = function () {
-            return null;
-        };
-
-        session.handleFileDownload = function () {
+        session.getAuthCredentials = () => null;
+        session.handleFileDownload = () => {
```
Comment (LavrovArtem) [inline test/server/external-proxy-test.js]: ```() => void 0```

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
FPR

**Pull request merged**